app/vmbackup/main.go

```diff
@@ -49,6 +49,11 @@ func main() {
 	logger.Init()
 	pushmetrics.Init()
 
+	// Storing snapshot delete function to be able to call it in case
+	// of error since logger.Fatal will exit the program without
+	// calling deferred functions.
+	var deleteSnapshot func()
```

Comment:
This code could panic if deleteSnapshot is not initialized and hide actual error from user.

 Possible solutions:
1) define no-op function `deleteSnapshot := func(){}`
2) define custom panic function, 
```go
reportFatalf(format string, args ...interface{}){
    if deleteSnapshot != nil {
        err := deleteSnapshot(); err != nil {
              format += fmt.Sprintf(" cannot delete snapshot on error: %s", err)
        }
    }
    // report actual error
     logger.Fatalf(format,args... )
  }

  ...
srcFS, err := newSrcFS()
	if err != nil {
           reportFatalf("%s", err)
}
```

Reply:
Properly initialized deleteSnapshot() func at 9436ae3b07bfa53f6d8b687a1ea1f9a903c94252